### ISSUE TITLE
Missing or black images in optimized PDFs

### ISSUE THREAD

**Issue by rbrito** [pts/pdfsizeopt#29]
Hi.

I have many files (I can send them privately) that show two (related/unrelated?) problems:

* the images go completely missing
* the images get black

Here is a link to one file that shows the problem: http://cs.stanford.edu/people/nick/compdocs/Practical_HI_Examples.pdf

For an second PDF file, I inserted some asserts and saw that the output below shows that the PS file (and the consequent PNG file) that is generated by ghostscript is already black:

```
info: This is pdfsizeopt rUNKNOWN size=349935.
info: prepending to PATH: /home/rbrito/Downloads/pdfsizeopt
info: loading PDF from: parallel-sieve-of-erathostenes.pdf
info: loaded PDF of 241349 bytes
info: separated to 95 objs + xref + trailer
info: found 0 Type1 fonts loaded
info: found 0 Type1C fonts loaded
info: will optimize image XObject 58; orig width=312 height=237 colorspace=/Indexed/DeviceGray bpc=8 filter=None dp=0 size=74362 gs_device=pnggray
info: optimizing 1 images of 74362 bytes in total
info: writing ImageRenderer (74387 image bytes) to: psotmp.9124.conv.pnggray.tmp.ps
info: using Ghostscript gs: GPL Ghostscript 9.21 (2017-03-16)
info: executing ImageRenderer with Ghostscript: gs -q -P- -dNOPAUSE -dBATCH -sDEVICE=pnggray -sOutputFile='psotmp.9124.img-%04d.pnggray.tmp.png' -f psotmp.9124.conv.pnggray.tmp.ps
ImageRenderer: rendering image XObject 58 width=312 height=237 bpc=8 colorspace=[/Indexed /DeviceGray] filter=null decodeparms=null device=pnggray
ImageRenderer: all OK
Traceback (most recent call last):
  File "/home/rbrito/Downloads/pdfsizeopt/pdfsizeopt", line 37, in <module>
    sys.exit(main.main(sys.argv))
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 8609, in main
    pdf.OptimizeImages(use_pngout=use_pngout, use_optipng=use_optipng, use_jbig2=use_jbig2)
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 6710, in OptimizeImages
    assert False
AssertionError
```

(This is a patched version to use optipng, but I am not using it here).

I'm attaching the PDF, and the respective PS and PNG files.

[parallel-sieve-of-erathostenes.pdf](https://github.com/pts/pdfsizeopt/files/1247663/parallel-sieve-of-erathostenes.pdf)
[psotmp.9124.conv.pnggray.tmp.ps.gz](https://github.com/pts/pdfsizeopt/files/1247667/psotmp.9124.conv.pnggray.tmp.ps.gz)
![psotmp 9124 img-0001 pnggray tmp](https://user-images.githubusercontent.com/212145/29648599-82e06e16-8865-11e7-9371-68759f616a03.png)

Thanks in advance,

Rogério Brito.

**Comment by pts** [comment]
Here is page 8 of parallel-sieve-of-erathostenes.pdf, which contains the offending image:
[pa8.pdf](https://github.com/pts/pdfsizeopt/files/1249364/pa8.pdf)

**Comment by pts** [comment]
Thank you for reporting this bug and sharing the example files!

Fixed in 55a87bd528f67f4a5e5d26c4eb9fad5e9ad89e55. Please upgrade to the latest pdfsizeopt to enjoy the correct processing.

If you find any more bugs, please report them as separate issues.

**Comment by rbrito** [comment]
Sorry for sending more than one file in a single issue. And thank you for fixing this issue. I do have more files that show errors and I will report them now.
